core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -77,17 +106,50 @@ class SubmissionMonitorActor(val workspaceName: WorkspaceName,
       logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
       if (terminateActor) stop(self)
       else scheduleNextMonitorPass
+    case CheckCurrentWorkflowStatusCounts =>
+      logger.debug(s"check current workflow status counts for submission $submissionId")
+      checkCurrentWorkflowStatusCounts pipeTo self
+    case SaveCurrentWorkflowStatusCounts(statusCounts) =>
+      logger.debug(s"saving current workflow status counts for submission $submissionId")
+      currentWorkflowStatusCounts = statusCounts
+      scheduleNextCheckCurrentWorkflowStatus
 
     case Status.Failure(t) => throw t // an error happened in some future, let the supervisor handle it
   }
 
-  def scheduleNextMonitorPass: Cancellable = {
+  private def scheduleNextMonitorPass: Cancellable = {
     system.scheduler.scheduleOnce(submissionPollInterval, self, StartMonitorPass)
   }
 
+  private def scheduleNextCheckCurrentWorkflowStatus: Cancellable = {
+    system.scheduler.scheduleOnce(addJitter(submissionPollInterval), self, CheckCurrentWorkflowStatusCounts)
+  }
+
+  private def initGauges(): Unit = {
+    try {
+      WorkflowStatuses.allStatuses.foreach { status =>
+        ExpandedMetricBuilder
+          .expand(WorkspaceMetric, workspaceName)
+          .expand(SubmissionMetric, submissionId)
```

Comment:
Yeah probably could reuse that. Though I don't think building the metric names is too expensive. :)